Cubrid Plugin Extension

longhaseng52 opened the pull request:
- Driver download from maven repository
- Connection properties
- Create folders to categorized by user
- Display list of user
- Display table based on user
- Display system table
- Display view based on user
- Display system view

longhaseng52 committed:
Cubrid Plugin Extension
- Create Cubrid plugin
- Edit connection and driver properties

longhaseng52 committed:
Cubrid Plugin Extension
- display list of user
- create folders to catagorized by user

longhaseng52 committed:
Cubrid Plugin Extension
- display list of table
- display list of system table

longhaseng52 committed:
Cubrid Plugin Extension
- display list of view
- display list of system view

hwany7seo commented:
Hello @longhaseng52  
PR is really great. But I have a list that I need to check.
would like you to check list.

[check list]
1.  Exception on Navigator -> Create Table
![image](https://github.com/CUBRID/dbeaver/assets/68524199/3558a406-c9da-4004-b2ea-bc760e1f5880)

2.  System Table&View should only be  visible in 'DBA' user

3. In Server 11.2 and later, column in tables appear duplicated.
ex) dba.test2
CREATE TABLE [dba].[test2] (
	[aaa] CHARACTER(1),
	[bbb] CHARACTER(1),
	[ggg] INTEGER
)
public.test2
CREATE TABLE [public].[test2] ([bbb] INTEGER)
![image](https://github.com/CUBRID/dbeaver/assets/68524199/0ceb7f78-b23d-42bf-a40f-a3e86cba7a36)

4. If a procedure is not retrieved, it must be removed from the list

rathana-pvs commented:
> 3. In Server 11.2 and later, column in tables appear duplicated.
>    ex) dba.test2
>    CREATE TABLE [dba].[test2] (
>    [aaa] CHARACTER(1),
>    [bbb] CHARACTER(1),
>    [ggg] INTEGER
>    )
>    public.test2
>    CREATE TABLE [public].[test2] ([bbb] INTEGER)
>    ![image](https://user-images.githubusercontent.com/68524199/284819671-0ceb7f78-b23d-42bf-a40f-a3e86cba7a36.png)

We have identified this issue. This issue is related to TableColumn. to fix this issue we need to modify some  of column. 
My opinion is: to avoid duplicate task, Can we fix this issue on third PR? 
because in the third PR we are going to focus column implementation.

hwany7seo commented:
> We have identified this issue. This issue is related to TableColumn. to fix this issue we need to modify some of column. My opinion is: to avoid duplicate task, Can we fix this issue on third PR? because in the third PR we are going to focus column implementation.

I wish it was included now, but I'll accept your opinion if so.

longhaseng52 committed:
Cubrid Plugin Extension
- Fixed create table with UI
- Fixed duplicated column
- Remove system table/view from other user
- Remove unimplemented procedure

longhaseng52 commented:
Dear @hwany7seo, we have fixed all the issues above including "column in tables appear duplicated". Please kindly check if there is still any issues. Thank you.

hwany7seo commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/edit/CubridTableManager.java:
```diff
@@ -0,0 +1,58 @@
+	private static final String NEW_TABLE_NAME = "NEW_TABLE";
+	private static final String TABLE_TYPE_TABLE = "TABLE";
+	private static final Class<? extends DBSObject>[] CHILD_TYPES = CommonUtils.array(
+	        GenericTableColumn.class,
```
Shouldn’t Generic... be CUBRID...?

hwany7seo reviewed (commented):
Other issues have been resolved. But a added new issue.

[New Issue]

1.  Columns information can be viewed in 'Navigator', 'Properties' editor  And on Server 11.0
Then, press F5 to refresh and it will be displayed.

test) Click Columns or Double Click 'athlete' in Navigator List

2.  When the table name(DBA.test2, PUBLIC.test2) is the same, PK, FK, and INDEX information are shared. ON Server 11.2

ex) 
CREATE TABLE DBA.[test2] (
	[aaa] CHARACTER(1),
	[bbb] CHARACTER(1),
	[ggg] INTEGER NOT NULL,
	[custkey] INTEGER,
	CONSTRAINT [pk] PRIMARY KEY([ggg]),
	INDEX [i_test2_bbb] ([bbb]),
	CONSTRAINT [fk_custkey] FOREIGN KEY([custkey]) REFERENCES [dba.customer] ([c_custkey]) ON DELETE RESTRICT ON UPDATE RESTRICT
)

CREATE TABLE PUBLIC.[test2] 
(
	[bbb] INTEGER
)

longhaseng52 committed:
Cubrid Plugin Extension
- Fixed column not loading on server 11.0
- Fixed duplicated primary key, foreign key and index

hwany7seo reviewed (approved):
Thank you for hard works.

kisoo-han commented:
@longhaseng52 ,
I'm also appreciate that. I checked both issues work fine.
